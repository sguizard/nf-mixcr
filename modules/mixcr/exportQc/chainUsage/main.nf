process MIXCR_EXPORTQC_CHAINUSAGE {
    tag "$study"
    container 'ghcr.io/milaboratory/mixcr/mixcr:4.6.0'

    input:
    val(study) 
    path(clns)
    path(library)

    output:
    path("*_exportQC_chainUsage.pdf"), emit: qc_chainusage_pdf
    path("*_exportQC_chainUsage.png"), emit: qc_chainusage_png
    path "versions.yml"              , emit: versions

    script:
    """
    export _JAVA_OPTIONS="-XX:-UsePerfData"

    mixcr exportQc chainUsage *.clns ${study}_exportQC_chainUsage.pdf
    mixcr exportQc chainUsage *.clns ${study}_exportQC_chainUsage.png

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """

    stub:
    """
    touch ${study}_exportQC_chainUsage.pdf
    touch ${study}_exportQC_chainUsage.png

    cat <<-END_VERSIONS > versions.yml
    MIXCR_ANALYZE:
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """
}
