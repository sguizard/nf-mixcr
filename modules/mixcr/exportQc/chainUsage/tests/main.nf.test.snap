{
    "MIXCR_EXPORTQC_CHAINUSAGE Compute chainusage QC": {
        "content": [
            {
                "0": [
                    "TCR_cats_exportQC_chainUsage.pdf:md5,09a39fe51a341845b9065fb43ab002cd"
                ],
                "1": [
                    "TCR_cats_exportQC_chainUsage.png:md5,99e4ad5ac697cf96ed4449b14fccd71b"
                ],
                "2": [
                    "versions.yml:md5,286ad196fbf69a87a1e99a2a55f65f89"
                ],
                "qc_chainusage_pdf": [
                    "TCR_cats_exportQC_chainUsage.pdf:md5,09a39fe51a341845b9065fb43ab002cd"
                ],
                "qc_chainusage_png": [
                    "TCR_cats_exportQC_chainUsage.png:md5,99e4ad5ac697cf96ed4449b14fccd71b"
                ],
                "versions": [
                    "versions.yml:md5,286ad196fbf69a87a1e99a2a55f65f89"
                ]
            }
        ],
        "timestamp": "2024-01-22T12:04:19.440227"
    }
}